src/screenipy.py

```diff
@@ -89,35 +68,117 @@ def initExecution():
+                processedData, screeningDictionary, saveDictionary, percentage=ConfigManager.consolidationPercentage)
+            isMaReversal = Screener.tools.validateMovingAverages(
+                processedData, screeningDictionary, saveDictionary, range=1.25)
+            isVolumeHigh = Screener.tools.validateVolume(
+                processedData, screeningDictionary, saveDictionary, volumeRatio=ConfigManager.volumeRatio)
+            isBreaking = Screener.tools.findBreakout(
+                processedData, screeningDictionary, saveDictionary, daysToLookback=ConfigManager.daysToLookback)
+            isLtpValid = Screener.tools.validateLTP(
+                fullData, screeningDictionary, saveDictionary, minLTP=ConfigManager.minLTP, maxLTP=ConfigManager.maxLTP)
+            isLowestVolume = Screener.tools.validateLowestVolume(
+                processedData, daysForLowestVolume)
+            isValidRsi = Screener.tools.validateRSI(
+                processedData, screeningDictionary, saveDictionary, minRSI, maxRSI)
+            currentTrend = Screener.tools.findTrend(
+                processedData, screeningDictionary, saveDictionary, daysToLookback=ConfigManager.daysToLookback, stockName=stock)
+            isCandlePattern = candlePatterns.findPattern(
+                processedData, screeningDictionary, saveDictionary)
+            isInsideBar = Screener.tools.validateInsideBar(
+                processedData, screeningDictionary, saveDictionary, bullBear=respBullBear, daysToLookback=insideBarToLookback)
+
+            if executeOption == 0:
+                return screeningDictionary, saveDictionary
+            if (executeOption == 1 or executeOption == 2) and isBreaking and isVolumeHigh and isLtpValid:
+                return screeningDictionary, saveDictionary
+            if (executeOption == 1 or executeOption == 3) and (consolidationValue <= ConfigManager.consolidationPercentage and consolidationValue != 0) and isLtpValid:
+                return screeningDictionary, saveDictionary
+            if executeOption == 4 and isLtpValid and isLowestVolume:
+                return screeningDictionary, saveDictionary
+            if executeOption == 5 and isLtpValid and isValidRsi:
+                return screeningDictionary, saveDictionary
+            if executeOption == 6 and isLtpValid:
+                if reversalOption == 1:
+                    if saveDictionary['Pattern'] in CandlePatterns.reversalPatternsBullish or isMaReversal > 0:
+                        return screeningDictionary, saveDictionary
+                elif reversalOption == 2:
+                    if saveDictionary['Pattern'] in CandlePatterns.reversalPatternsBearish or isMaReversal < 0:
+                        return screeningDictionary, saveDictionary
+            if executeOption == 7 and isLtpValid and isInsideBar:
+                return screeningDictionary, saveDictionary
+    except KeyboardInterrupt:
```

Comment:
The processes created by `pool` are **not terminating even on a `KeyboardInterrupt`**.
This is a bug that needs to be fixed. user should be able to kill the screening processes by pressing `Ctrl+C` and should get the results that are stored in `screenResults` dataframe till that time.

Reply:
Ok, I will work on it.

Reply:
`Pool` did not worked. Switched to `Process` and **`KeyboardInterrupt`** is handled gracefully in ea9497e